presto-jdbc/src/main/java/io/prestosql/jdbc/AbstractPrestoResultSet.java

```diff
@@ -228,7 +230,14 @@ public String getString(int columnIndex)
             throws SQLException
     {
         Object value = column(columnIndex);
-        return (value != null) ? value.toString() : null;
+        if (value == null) {
+            return null;
+        }
+        ClientTypeSignature columnTypeSignature = columnInfo(columnIndex).getColumnTypeSignature();
+        if (TYPE_CONVERSIONS.hasConversion(columnTypeSignature.getRawType(), String.class)) {
+            return TYPE_CONVERSIONS.convert(columnTypeSignature, value, String.class);
+        }
+        return value.toString();
```

Comment:
This is of course not useful for any other type backed by `byte[]`.
Perhaps we could realize value is an array and use `java.util.Arrays#deepToString` here.